iroh-dns-server/src/config.rs

```diff
@@ -173,7 +173,7 @@ impl Config {
             PathBuf::from(val)
         } else {
             let path = dirs_next::data_dir()
-                .context("operating environment provides no directory for application data")?;
+                .std_context("operating environment provides no directory for application data")?;
```

Comment:
I think this works reasonably well, a bit confusing at first, but easy to use at the end of the day

Reply:
To me, this is the most annoying thing about this whole endevour, but unfortunately I don't think there's another solution. I made the `std_context` method the "longer" one (to type) to nudge towards using `context` wherever possible (ie. on our errors that implement StackError) because `std_context` will work on our errors too, but we'll lose the location info in the process. This is the much discussed issue that as long as we implement std::error::Error for our errors, we can't forbid calling the StdResultExt  methods on them, even though we *should*  always use the StackResultExt methods (which are only available on StackError results, not on std error results).